apps/devhub/views.py

```diff
@@ -621,6 +621,8 @@ def handle_upload(filedata, user, app_id=None, version_id=None, addon=None,
         is_listed = addon.is_listed
 
     fu = FileUpload.from_post(filedata, filedata.name, filedata.size)
+    if addon is not None:
+        fu.addon = addon
```

Comment:
`from_post()` seems to just deal with the actual file and we add some more data to it once it is returned below (`automated_signing` and `user`). So, yes it feels like it should go somewhere else but it seems to be designed this way.

The `FileUpload` also comes back unsaved so we at least need to save/update it. Setting the addon to `None` should be fine so I merged this with the next line.

```py
fu.update(automated_signing=automated, addon=addon)